app/actions/remote/channel.ts

```diff
@@ -49,6 +51,98 @@ export type MyChannelsRequest = {
+
+        return {error: undefined};
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error as ClientErrorProps);
+        return {error};
+    }
+}
+
+export async function fetchChannelMembersByIds(serverUrl: string, channelId: string, userIds: string[], fetchOnly = false): Promise<ChannelMembersRequest> {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const members = await client.getChannelMembersByIds(channelId, userIds);
+
+        if (!fetchOnly) {
+            const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+            if (operator && members.length) {
+                const modelPromises: Array<Promise<Model[]>> = [];
+                const membership = members.map((u) => ({
+                    channel_id: channelId,
+                    user_id: u.user_id,
+                    scheme_admin: u.scheme_admin,
+                }));
+                modelPromises.push(operator.handleChannelMembership({
+                    channelMemberships: membership,
+                    prepareRecordsOnly: true,
+                }));
+
+                const models = await Promise.all(modelPromises);
+                await operator.batchRecords(models.flat());
+            }
+        }
+
+        return {members};
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error as ClientError);
+        return {error};
+    }
+}
+export async function updateChannelMemberSchemeRoles(serverUrl: string, channelId: string, userId: string, isSchemeUser: boolean, isSchemeAdmin: boolean, fetchOnly = false) {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        await client.updateChannelMemberSchemeRoles(channelId, userId, isSchemeUser, isSchemeAdmin);
+
+        if (!fetchOnly) {
+            return await getMemberInChannel(serverUrl, channelId, userId);
+        }
+
+        return {error: undefined};
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error as ClientErrorProps);
+        return {error};
+    }
+}
+
+export async function getMemberInChannel(serverUrl: string, channelId: string, userId: string, fetchOnly = false) {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const member = await client.getMemberInChannel(channelId, userId);
+
+        if (!fetchOnly) {
+            await updateLocalUser(serverUrl, member, userId);
+        }
+        return {member, error: undefined};
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error as ClientErrorProps);
+        return {error};
+    }
+}
+
+export async function fetchChannelMemberships(serverUrl: string, channelId: string, options: GetUsersOptions, fetchOnly = false) {
+    const {users = []} = await fetchProfilesInChannel(serverUrl, channelId, undefined, options, fetchOnly);
+    const userIds = users.map((u) => u.id);
+
+    // We are not sure the getChannelMembers API returns the same members
```

Comment:
Why? Is this a hack or should a ticket be submitted to Server team for them to implement it?

Reply:
created ticket and added comment with ticket number